### PR TITLE
Add new "Pipelines" page to the sidebar of SCT's RTD documentation

### LINKED ISSUE #3621: Add link to sct-pipeline in the documentation
There are some useful pipelines, that others could benefit from, here: https://github.com/sct-pipeline

My suggestion would be to add a section 'Pipelines' right below 'command line tools':

image

This section would have a link to the GH organization sct-pipeline.

### PR DESCRIPTION
**Pull request by RignonNoel**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://intranet.neuro.polymtl.ca/geek-tips/contributing#pr-labels-a-href-pr-labels-id-pr-labels-a) to this PR
- [ ] I've applied a [release milestone](https://github.com/spinalcordtoolbox/spinalcordtoolbox/milestones) (major, minor, patch) in line with [Semantic Versioning guidelines](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Misc%3A-Creating-a-new-release#convention-for-naming-releases) 
- [x] I've assigned a reviewer

#### PR contents

- [x] I've consulted [SCT's internal developer documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [x] I've added [relevant tests](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Tests) for my contribution
- [x] I've updated the [relevant documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description
Added a new `pipelines` section to the documentation to reference the work done on SCT-pipeline github organization and help people find some useful example and contributions.

## Linked issues
Fixes #3621

### CODE REVIEW

File: documentation/source/user_section/pipelines.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Pipelines
+#########
+
+SCT maintain the `sct-pipeline <https://github.com/sct-pipeline>`_ Github organization to references some useful pipelines compatible with the Spinal Cord Toolbox (SCT) that you could benefit from.
```
Comment (jcohenadad) [inline documentation/source/user_section/pipelines.rst:4]:
```suggestion
SCT team maintains the `sct-pipeline <https://github.com/sct-pipeline>`_ Github organization to references some useful pipelines compatible with SCT that anyone could benefit from.
```

File: documentation/source/user_section/pipelines.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Pipelines
+#########
+
+SCT team maintains the `sct-pipeline <https://github.com/sct-pipeline>`_ Github organization to references some useful pipelines compatible with the Spinal Cord Toolbox (SCT) that anyone could benefit from.
```
[the remark this answers is not in the document]
Comment (joshuacwnewton) [reply documentation/source/user_section/pipelines.rst:4]: (Note: I'm not sure if these are the best repositories to highlight here -- perhaps @jcohenadad could chime in about whether there are better `sct-pipeline` repositories worth mentioning.)

**Review by joshuacwnewton — COMMENTED** [review]
Thank you for taking this on! :tada:

File: documentation/source/user_section/pipelines.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (jcohenadad) [reply documentation/source/user_section/pipelines.rst:4]: @joshuacwnewton your suggestion is excellent, this is exactly the kind of context that was missing

File: documentation/source/user_section/pipelines.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Pipelines
+#########
```
[the remark this answers is not in the document]
Comment (jcohenadad) [reply documentation/source/user_section/pipelines.rst:2]: "Analysis pipelines" might reduce the possible confusion for some users.  I'm not sure about "Applications of SCT", as it _sort of_ overlap with the citation list of examples of applications of SCT (except that these are repos with code, not papers, but in the mind of our users, that might be a bit confusing).

[the remark this answers is not in the document]
Comment (joshuacwnewton) [reply documentation/source/user_section/pipelines.rst:2]: Good point, there is definitely some overlap with the studies page. :thinking: 

I like Analysis Pipelines the most out of the three. :)

[the remark this answers is not in the document]
Comment (RignonNoel) [reply documentation/source/user_section/pipelines.rst]: I changed the name accordingly

**Comment by RignonNoel** [comment]
@joshuacwnewton I really like the proposition of text, I commited it.

Name of the page is renamed.
Branch is up-to-date with last change on master.

**Review by joshuacwnewton — APPROVED** [review]
LGTM! :tada:

**Comment by joshuacwnewton** [comment]
Merging this with admin privileges because we know that the test failure is due to #3675 which is solved by #3676.

**Pull request merged**